### PR TITLE
[PATCH v3] linux-dpdk: timer: change to rte_timer based implementation

### PR DESCRIPTION
**Pull request by psavol**
Use rte_timer to implement timer. Plain queue integration needs
to be improved from queue / queue interface side.

Signed-off-by: Petri Savolainen <petri.savolainen@linaro.org>

### CODE REVIEW

**Comment by Bill-Fischofer-Linaro** [comment]
For some reason the last four tests timed out with no explanation. When restarted they pass so Travis now seems happy with this PR.

File: platform/linux-dpdk/odp_timer.c [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+		if (timer_pool->used == 0) {
+			timer_pool->used = 1;
+			break;
+		}
+	}
+
+	odp_ticketlock_unlock(&timer_global->lock);
+
+	if (i == MAX_TIMER_POOLS) {
+		ODP_ERR("Too many timer pools\n");
```
Comment (Bill-Fischofer-Linaro) [inline platform/linux-dpdk/odp_timer.c]: Isn't "All timer pools already in use\n" the real error here? The issue isn't that there are too many timer pools, but that the application is requesting more timer pools than are available.

File: platform/linux-dpdk/odp_timer.c [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+		timer->tmo_event = ODP_EVENT_INVALID;
+		timer->active = 0;
+	} else {
+		ev = ODP_EVENT_INVALID;
+	}
+
+	odp_ticketlock_unlock(&timer->lock);
+
+	/* TODO: disable timer polling from queue. Needs better queue <-> timer
+	 * interface. */
```
Comment (Bill-Fischofer-Linaro) [inline platform/linux-dpdk/odp_timer.c]: Note this in TODO issue as well.

**Comment by muvarov** [comment]
is rte_timer faster than linux-generic timer implementation?

**Comment by psavol** [comment]
v2: rebase and implementation changes after multi-thread testing (odp_sched_pktio -t).

File: platform/linux-dpdk/odp_timer.c (hunk 1 above)
Comment (psavol) [reply platform/linux-dpdk/odp_timer.c]: Fixed in v3

**Comment by psavol** [comment]
v3: rebase and documentation corrections

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**